### ISSUE TITLE
openssl wrap doesn't build on windows (vs via ninja)

### ISSUE THREAD

**Issue by anarazel** [mesonbuild/wrapdb#181]
Hi,

```
PS C:\Users\anfreund\src\pg-meson\build-ninja-x64> ninja .\subprojects\openssl-1.1.1l\openssl.exe
[1/1] Linking target subprojects/openssl-1.1.1l/openssl.exe
FAILED: subprojects/openssl-1.1.1l/openssl.exe subprojects/openssl-1.1.1l/openssl.pdb
"link" @subprojects/openssl-1.1.1l/openssl.exe.rsp
LINK : fatal error LNK1181: cannot open input file 'subprojects\openssl-1.1.1l\crypto.lib'
```
As far as I can tell the reason for the failure to build is that the buildsystem contained in the wrap doesn't take care of exporting symbols. Due to that link.exe doesn't produce an import library, which then causes this issue.

CC: @nazar-pc

Regards,

Andres

**Comment by vinipsmaker** [comment]
Any updates on this?

**Comment by punytroll** [comment]
Good to know, that I'm not the only one with this problem!

@xclaesse You mentioned some things in your reply from nearly a year ago. Could you elaborate on the issues ... I am very willing to try my hands on this, but I could use some starters ...

Thanks! :)

**Comment by amyspark** [comment]
I have a version ready for public consumption, I'll update it to 1.1.1q and fix #428 before linking it here.

**Comment by amyspark** [comment]
@punytroll @jfinkhaeuser, I've made a custom wrap for OpenSSL 1.x based on the existing version for v3.

Mine improves the following bits:

- support for MSYS compilers (64 and 32-bit) as well as MSVC
- detection of the assemblers available (gas/nasm)
- adds a .pc file for easy lookup
- enabled the usage of nasm for MSVC

The latter was a longstanding TODO since https://github.com/mesonbuild/wrapdb/commit/7dfccc106223afe3447ca8b1eff06ef0b5fbc252.

You'll find the wrap in [this branch](https://github.com/amyspark/wrapdb/tree/openssl_1.1.1q-1) and the release [here](https://github.com/amyspark/wrapdb/releases/tag/openssl_1.1.1q-1). I'm not sure if/how I could upstream it here, given that wrapdb has already moved to the 3.x branch.

**Comment by jfinkhaeuser** [comment]
Thank you, I'll check it out!

**Comment by jfinkhaeuser** [comment]
@amyspark it doesn't quite work for me, but it's enough that I can probably work my way there. FWIW, I would have no problem with using v3, but then there's #428 and #427 in those.

At any rate, I have more info with this, so thank you very much!

**Comment by lemosjose** [comment]
Any udpates on that? got into similar stuff while building a project, an i see it's not closed yet

**Comment by hrxi** [comment]
Would be fixed by #1283.
